src/main/java/edu/harvard/iq/dataverse/util/SignpostingResources.java

```diff
@@ -0,0 +1,346 @@
+                        systemConfig.getDataverseSiteUrl() + "/api/datasets/export?exporter=schema.org&persistentId=" + ds.getProtocol() + ":" + ds.getAuthority() + "/" + ds.getIdentifier()
+                ).add(
+                        "type",
+                        "application/json+ld"
+                )
+        );
+        JsonArrayBuilder linksetJsonObj = Json.createArrayBuilder();
+
+        JsonObjectBuilder mandatory;
+        if (useDefaultFileType) {
+            mandatory = jsonObjectBuilder()
+                    .add("anchor", landingPage)
+                    .add("cite-as", Json.createArrayBuilder().add(jsonObjectBuilder().add("href", ds.getPersistentURL())))
+                    .add("type", Json.createArrayBuilder()
+                            .add(jsonObjectBuilder().add("href", "https://schema.org/AboutPage"))
+                            .add(jsonObjectBuilder().add("href", defaultFileTypeValue))
+                    );
+        } else {
+            mandatory = jsonObjectBuilder()
+                    .add("anchor", landingPage)
+                    .add("cite-as", Json.createArrayBuilder().add(jsonObjectBuilder().add("href", ds.getPersistentURL())))
+                    .add("type", Json.createArrayBuilder()
+                            .add(jsonObjectBuilder().add("href", "https://schema.org/AboutPage"))
+                    );
+        }
+
+        if (authors != null) {
+            mandatory.add("author", authors);
+        }
+        if (licenseString != null && !Objects.equals(licenseString, "")) {
+            mandatory.add("license", jsonObjectBuilder().add("href", licenseString));
+        }
+        if (!mediaTypes.toString().trim().isEmpty()) {
+            mandatory.add("describedby", mediaTypes);
+        }
+        if (items != null) {
+            mandatory.add("item", items);
+        }
+        linksetJsonObj.add(mandatory);
+
+        // remove scholarly type as shown already on landing page
+        for (FileMetadata fm : fms) {
+            DataFile df = fm.getDataFile();
+            JsonObjectBuilder itemAnchor = jsonObjectBuilder().add("anchor", getPublicDownloadUrl(df));
+            itemAnchor.add("collection", Json.createArrayBuilder().add(jsonObjectBuilder()
+                    .add("href", landingPage)));
+            linksetJsonObj.add(itemAnchor);
+        }
+
+        return linksetJsonObj;
+    }
+
+
+    private String getPublicDownloadUrl(DataFile dataFile) {
+        StorageIO<DataFile> storageIO = null;
+        try {
+            storageIO = dataFile.getStorageIO();
+        } catch (IOException e) {
+            logger.warning(String.format("Error getting storageID from file; original error message is: %s", e.getLocalizedMessage()));
+        }
+
+        if (storageIO instanceof SwiftAccessIO) {
+            String fileDownloadUrl;
+            SwiftAccessIO<DataFile> swiftIO = (SwiftAccessIO<DataFile>) storageIO;
+            try {
+                swiftIO.open();
```

Comment:
Not completely sure about swift, but other stores, once you do an open(), you have streams that have to be closed to avoid resource leaks.